apps/ejabberd/src/mod_roster_riak.erl

```diff
@@ -0,0 +1,192 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_roster_riak.erl
+%%% Author  : Piotr Nosek <piotr.nosek@erlang-solutions.com>
+%%% Purpose : mod_roster Riak backend
+%%%
+%%% MongooseIM, Copyright (C) 2015      Erlang Solutions Ltd.
+%%%
+%%%----------------------------------------------------------------------
+-module(mod_roster_riak).
+
+-include("mod_roster.hrl").
+-include("jlib.hrl").
+
+-behaviour(mod_roster).
+
+%% API
+-export([init/2,
+         transaction/1,
+         read_roster_version/2,
+         write_roster_version/4,
+         get_roster/2,
+         get_roster_by_jid_t/3,
+         get_subscription_lists/3,
+         roster_subscribe_t/4,
+         get_roster_by_jid_with_groups_t/3,
+         remove_user/2,
+         update_roster_t/4,
+         del_roster_t/3,
+         read_subscription_and_groups/3,
+         raw_to_record/2]).
+
+-define(ROSTER_BUCKET(LServer), {<<"rosters">>, LServer}).
+-define(VER_BUCKET(LServer), {<<"roster_versions">>, LServer}).
+
+-spec init(ejabberd:server(), list()) -> ok.
+init(_Host, _Opts) ->
+    ok. % Common Riak pool is used
+
+-spec transaction(F :: fun()) -> {aborted, Reason :: any()} | {atomic, Result :: any()}.
+transaction(F) ->
+    put(riak_roster_t, []),
+    put(riak_version_t, []),
+    Return = case catch F() of
+                 {'EXIT', Reason} ->
+                     {aborted, Reason};
+                 Result ->
+                     %% Commiting...
+                     commit_t_roster(),
```

Comment:
Are we sure that the `commit_t_roster()` and `commit_t_version()` always succeeds?